### DISCUSSION TITLE
Method to handle the opening of a direct FCM->notifications when app swiped up (Android)

### DISCUSSION THREAD

**Discussion by alelaru** [invertase/react-native-firebase#6620] [Q&A]
Android: When a user swipes up the app and gets a notification (FCM->notification) it shows the notification normally... Then, the user clicks on the notification and nothing happens.

I have onMessage, onNotificationOpenedApp and getInitialNotification.

Can someone explains to me which method should handle this event? As far as i understand is getInitialNotification but not sure. My method setBackgroundMessageHandler() gets the information when the notification arrives but that doesnt help me with my use case which is that i want to move the user to the main screen... Im not using the Splash package other issues show and i am currently in the latest version of the firebase packages + react + react-native

The problem have is that if the app is not swiped up, onMessage and and onNotificationOpenedApp will handle those cases but with the swipe up there is no information even in the logcat

**Comment by mikehardy** [comment]
Hi there!

You have specified no versions of software used to build your app or to execute this use case (https://stackoverflow.com/help/how-to-ask)

So all of my thoughts are guesses.

> When a user swipes up the app and gets a notification (FCM->notification) it shows the notification normally... Then, the user clicks on the notification and nothing happens.

really  nothing happens? I expect your app to boot and show it's user interface, yes?

It should be getIniitalNotification. If you search issues here, and my memory serves me, some people have had issues with calling getInitialNotification immediately on startup and have put a call in to that API with a short delay after app boot. This is a vague recollection on my part. I use getInitiailNotification though, and then navigate to some deep link screen depending on the notification, after app boot

**Reply by alelaru** [reply]
Thank you Mike i am using the latest versions:
- "@react-native-firebase/app": "14.11.1"
- "@react-native-firebase/messaging": "14.11.1"
- "react": "18.1.0"
- "react-native": "0.70.3"
-  Android 12
- notifee 5.6.0

My mistake, the app does open but not functions are called (I have several logs in all functions mentioned)

If i check the app's info and do a force stop, when i open the app a normal startup takes place (Takes few seconds and shows up an implemented spinner) and getInitial notification is called... Swiping up appears to do a different behavior because if i open the app it does it immediately, im not sure if in this case getInitialNotification will be called...

More info -> In the same app i also receive data messages from FCM which are handled by Notifee, when i do the same procedure and send a data message instead of a notification from FCM, Notifee shows the notification and gets a BackgroundEvent of type Delivered, when the user click the notification it gets an onForegroundEvent of type Press... very strange behavior i must say

As far as i understand react-native-firebase implementation does not differentiate between onForeground and onBackground right?

**Reply by mikehardy** [reply]
Hey - 

You are not using latest versions, these are definitely not latest:

```
"@react-native-firebase/app": "14.11.1"
"@react-native-firebase/messaging": "14.11.1"
```

I cannot speculate based on your descriptions, sorry. Nothing beats a single index.js that demonstrates how you install handlers then has a simple App component that shows behavior - https://stackoverflow.com/help/minimal-reproducible-example
